apps/baseline/serializers.py

```diff
@@ -1466,3 +1470,175 @@ def get_strategy_label(self, obj):
 
     def get_wealth_group_label(self, obj):
         return str(obj.wealth_group)
+
+
+class DictQuerySetField(rest_framework_fields.SerializerMethodField):
+    def __init__(self, field_name=None, **kwargs):
+        self.field_name = field_name
+        super().__init__(**kwargs)
+
+    def to_representation(self, obj):
+        return self.parent.get_field(obj, self.field_name)
+
+
+class LivelihoodZoneBaselineReportSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = LivelihoodZoneBaseline
+        fields = (
+            "id",
+            "name",
+            "description",
+            "source_organization",
+            "source_organization_name",
+            "livelihood_zone",
+            "livelihood_zone_name",
+            "country_pk",
+            "country_iso_en_name",
+            "main_livelihood_category",
+            "bss",
+            "currency",
+            "reference_year_start_date",
+            "reference_year_end_date",
+            "valid_from_date",
+            "valid_to_date",  # to display "is latest" / "is historic" in the UI for each ref yr
+            "population_source",
+            "population_estimate",
+            "livelihoodzone_pk",
+            "livelihood_strategy_pk",
+            "strategy_type",
+            "livelihood_activity_pk",
+            "wealth_group_category_code",
+            "population_estimate",
+            "slice_sum_kcals_consumed",
+            "sum_kcals_consumed",
+            "kcals_consumed_percent",
+            "product_cpc",
+            "product_common_name",
+        )
+
+    # For each of these aggregates the following calculation columns are added:
+    #   (a) Total at the LZB level (filtered by population, wealth group, etc), eg, sum_kcals_consumed.
+    #   (b) Total for the selected product/strategy type slice, eg, slice_sum_kcals_consumed.
+    #   (c) The percentage the slice represents of the whole, eg, kcals_consumed_percent.
+    # Filters are automatically created, eg, min_kcals_consumed_percent and max_kcals_consumed_percent.
+    # If no ordering is specified by the FilterSet, the results are ordered by percent descending in the order here.
+    aggregates = {
+        "kcals_consumed": Sum,
```

Comment:
what is preventing us from adding `quantity_produced`, `quantity_consumed`, `income`, `expenditure`, etc. - i.e. any field from `LivelihoodActivity`?

Reply:
I think that based on all the charts, we will need kcals_consumed, income and expenditure for most purposes - and a way to do the "complete" graph that converts income and expenditure to kcals.